Use unique variable name for Logger instance

KIVagant opened the pull request:
`log` is used in `for log in logs` loops where its use is more justified.

Signed-off-by: Eugene Glotov <kivagant@gmail.com>

<!--- Please remember to review the [contribution guidelines](https://github.com/DataDog/datadog-serverless-functions/blob/master/CONTRIBUTING.md) if you have not yet done so._  --->

### What does this PR do?

<!--- A brief description of the change being made with this pull request. --->

### Motivation

Using the same variable name for different purposes within one file is confusing. Especially when one of the variables is intended for global use.

### Testing Guidelines

The logic has not been touched.

### Additional Notes

nothing

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests

tianchu commented:
@KIVagant thanks for the contribution, indeed, `log` is a confusing variable name in the given context. Do you mind reformatting the `lambda_function.py` with `black` to make the linter pass? 

BTW, do you mind applying the change to `enhanced_lambda_metrics.py` as well?

KIVagant commented:
>  Do you mind reformatting the lambda_function.py with black to make the linter pass?

Sure, I'll take a look tomorrow.

KIVagant committed:
Use unique variable name for Logger instance

`log` is used in `for log in logs` loops where its use is more justified.

Also this fixes a bug in `enhanced_lambda_metrics.py:parse_and_submit_enhanced_metrics()` related to the naming conflict.

Signed-off-by: Eugene Glotov <kivagant@gmail.com>

KIVagant commented:
@tianchu , I updated the code. Thanks for mentioning the other file, there was a bug in `enhanced_lambda_metrics.py:parse_and_submit_enhanced_metrics()` related to the naming conflict.

tianchu reviewed (approved):
@KIVagant Thanks for fixing the bug and your contribution, this looks great!
